application/src/main/java/org/cardanofoundation/ledgersync/listeners/BlockEventListener.java

```diff
@@ -109,6 +121,10 @@ public void handleRollback(RollbackEvent rollbackEvent) {
         rollbackService.rollBackFrom(rollbackBlockNo);
         metricCollectorService.collectRollbackMetric();
         blockCount.set(0);
+
+        healthCheckCachingService.saveLatestBlockSlot(rollbackEvent.getRollbackTo().getSlot());
+        healthCheckCachingService.saveLatestBlockInsertTime(LocalDateTime.now(ZoneOffset.UTC));
+        healthCheckCachingService.saveLatestBlockTime(rollBackBlock.get().getTime().toLocalDateTime());
```

Comment:
Do we need to use Zone (UTC) like it's done for genesis block above?

Or, should we just keep latestBlockTime as long?  AFAIK, blockTime is actually in seconds. So, we can get the current time in seconds and find the difference.

Reply:
In this case, toLocalDateTime returns the correct value, rollBackBlock.get().getTime() returns the Timestamp value that was previously stored with UTC+0
We could keep latestBlockTime as long, but in rollback case as above, we cannot save a correct "long" value for latest block time